File: src/test/java/com/dotnomi/fabricdependencyinjection/testclasses/maininjection/MainDependentService.java
package com.dotnomi.fabricdependencyinjection.testclasses.maininjection;

import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
public class MainDependentService {
  private final MainInjectionTestModMain mainInstance;

  @ModInject
  public MainDependentService(MainInjectionTestModMain mainInstance) {
    this.mainInstance = mainInstance;
  }

  public MainInjectionTestModMain getMainInstance() { return mainInstance; }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/maininjection/MainDependentService.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.maininjection

import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class MainDependentService @ModInject constructor(val mainInstance: MainInjectionTestModMain)
